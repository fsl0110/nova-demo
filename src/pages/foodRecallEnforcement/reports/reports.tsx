import React, { FC, HTMLAttributes } from "react";
import { connect } from "react-redux";
import classNames from "classnames";
import { Text, Search, ChartHighstock } from "../../../components";
import { AppState } from "../../../typings";
import { fetchReports } from "../../../store/actions";
import text from "../../../static/data/text.json";

export interface StateProps {
  /** Is loading data. */
  isLoading: boolean;
  /** Has an error. */
  hasError: boolean;
  /** Options and Data of the chart */
  chartOptions: Highcharts.Options;
}

export interface DispatchProps {
  /** Action for fetching data. **/
  fetchReports: () => void;
}

export interface OwnProps {}

export type Props = StateProps &
  DispatchProps &
  OwnProps &
  HTMLAttributes<HTMLDivElement>;

export const Reports: FC<Props> = ({
  isLoading,
  hasError,
  chartOptions,
  fetchReports,
  className
}) => (
  <div className={classNames(className)}>
    <Text topicTag="h1" topic={text.reports.topic} text={text.reports.text} />
    <Search fetchData={fetchReports} isLoading={isLoading} />
    <div>
      <h3 className="horizontal-center">Enforcement reports over time</h3>
      <ChartHighstock
        className="chart-wrapper"
        isLoading={isLoading}
        hasError={hasError}
        chartOptions={chartOptions}
        fetchHandler={fetchReports}
      />
    </div>
  </div>
);

export const ReportsConnected = connect(
  (state: AppState): StateProps => ({
    isLoading: state.reports.isLoading,
    hasError: state.reports.hasError,
    chartOptions: state.reports.chartOptions
  }),
  {
    fetchReports
  }
)(Reports);
