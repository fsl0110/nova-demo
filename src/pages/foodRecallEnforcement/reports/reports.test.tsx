import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import moment from "moment";
import { Reports, Props } from "./reports";
import { findByTestAttr } from "../../../utils/tests";

const testProps: any = {
  fetchData: jest.fn(),
  chartOptions: {
    chart: {
      height: 350,
      style: {
        fontFamily:
          "-apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Oxygen"
      }
    },
    credits: {
      enabled: false
    },
    loading: {
      hideDuration: 500,
      showDuration: 300
    },
    title: {
      text: ""
    },
    yAxis: {
      title: {
        text: null
      }
    },
    xAxis: {
      type: "datetime",

      minTickInterval: moment.duration(1, "month").asMilliseconds()
    },
    series: [{ type: "line", data: [null] }]
  }
};
describe("<Text>", () => {
  it("renders toMatchSnapshot", () => {
    expect(toJson(shallow(<Reports {...testProps} />))).toMatchSnapshot();
  });
});

describe("<Components>", () => {
  let wrapper: any;
  let mockFunc: any;

  beforeEach(() => {
    mockFunc = jest.fn();
    const testProp: any = {
      fetchData: mockFunc,
      fetchStart: mockFunc,
      chartOptions: {
        chart: {
          height: 350,
          style: {
            fontFamily:
              "-apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Oxygen"
          }
        },
        credits: {
          enabled: false
        },
        loading: {
          hideDuration: 500,
          showDuration: 300
        },
        title: {
          text: ""
        },
        yAxis: {
          title: {
            text: null
          }
        },
        xAxis: {
          type: "datetime",

          minTickInterval: moment.duration(1, "month").asMilliseconds()
        },
        series: [{ type: "line", data: [null] }]
      }
    };
    wrapper = shallow(<Reports {...testProp} />);
  });

  it("Render Search", () => {
    const search = findByTestAttr(wrapper, "search");
    expect(search.length).toBe(1);
  });

  it("Render Text", () => {
    const text = findByTestAttr(wrapper, "text");
    expect(text.length).toBe(1);
  });

  it("Render ChartHighstock", () => {
    const chart = findByTestAttr(wrapper, "chart");
    expect(chart.length).toBe(1);
  });
});
