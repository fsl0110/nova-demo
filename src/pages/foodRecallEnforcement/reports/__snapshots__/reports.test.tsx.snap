// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text> renders toMatchSnapshot 1`] = `
<div
  className=""
  data-test="Reports"
>
  <Text
    data-test="text"
    text="This is the openFDA API endpoint for all food product recalls monitored by the FDA. When an FDA-regulated product is either defective or potentially harmful, recalling that product—removing it from the market or correcting the problem—is the most effective means for protecting the public."
    topic="Food recall enforcement reports since 2012"
    topicTag="h1"
  />
  <Search
    data-test="search"
    fetchData={[MockFunction]}
    location="reports"
  />
  <div>
    <h3
      className="horizontal-center"
    >
      Enforcement reports over time
    </h3>
    <ChartHighstock
      chartOptions={
        Object {
          "chart": Object {
            "height": 350,
            "style": Object {
              "fontFamily": "-apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Oxygen",
            },
          },
          "credits": Object {
            "enabled": false,
          },
          "loading": Object {
            "hideDuration": 500,
            "showDuration": 300,
          },
          "series": Array [
            Object {
              "data": Array [
                null,
              ],
              "type": "line",
            },
          ],
          "title": Object {
            "text": "",
          },
          "xAxis": Object {
            "minTickInterval": 2592000000,
            "type": "datetime",
          },
          "yAxis": Object {
            "title": Object {
              "text": null,
            },
          },
        }
      }
      className="chart-wrapper"
      data-test="chart"
      fetchHandler={[MockFunction]}
      location="reports"
    />
  </div>
</div>
`;
